### ISSUE TITLE
EoC: spell targets can't assign alpha or beta talker

### ISSUE THREAD

**Issue by GuardianDll** [CleverRaven/Cataclysm-DDA#70169]
### Describe the bug

at least it looks like so: i tried to make a spell before, that target all critters EoC around me, and EoC that teleports this critters, but it didn't work
Luckily, i think i have a better example of it

### Attach save file

N/A

### Steps to reproduce

Make a world with xedra evolved mod
Spawn `constructed hammer`, activate it, cast a spell anywhere around
Find that hammer is not despawned, despite spell `constructed_hammer_attack` casts `constructed_hammer_bash`, which then casts `constructed_hammer_self_destruct`, which then runs `EOC_REMOVE_CONSTRUCTED_HAMMER`, which effect is
```json
  {
    "id": "EOC_REMOVE_CONSTRUCTED_HAMMER",
    "type": "effect_on_condition",
    "effect": [ { "u_remove_item_with": "constructed_hammer" } ]
  },
  ```

### Expected behavior

Spell being able to properly assign alpha/beta talkers to spell, as [doc](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/EFFECT_ON_CONDITION.md#typical-alpha-and-beta-talkers-by-cases) mentions it
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/67688115/e8d8f183-3bdf-4828-a10d-d84c1ff6fc7f)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3693 (22H2)
- Game Version: 5fef522 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Slowdown Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills],
    Magiclysm [magiclysm],
    Xedra Evolved [xedra_evolved]
]

**Comment by PipeYume** [comment]
If you directly cast the spell "constructed_hammer_self_destruct" in the spell ui, the hammer can be destructed correctly.

If you use the hammer to cast that spell directly, I mean:
`"use_action": { "type": "cast_spell", "spell_id": "constructed_hammer_self_destruct", "no_fail": true, "level": 0, "need_wielding": true }`
There will be a segment fault, because you can destroy an item that is casting spells. The EOC still works. 

In conclusion, I guess there are something wrong with "extra_effects", so the alpha and beta talker for that eoc is wrong.
